### PR TITLE
Add "get_device_by_name" helper

### PR DESCRIPTION
**Pull request by shawnc722**
Takes a string name, searches the devices for one matching that name, and returns that device. If given an invalid name, raises a ValueError.

### CODE REVIEW

**Comment by jath03** [comment]
There will often be multiple devices with the same name (especially with RGB RAM), so I'd want this to return a list of devices, similar to the `get_devices_by_type` function.  Otherwise, it would only ever return the first device of a certain name.

**Comment by shawnc722** [comment]
I checked with my own setup and saw that all my names are unique (my RAM sticks are each zones of a shared device) and assumed it'd be that way for everyone. Updated the helper to make it "get_devices_by_name" so it returns a list, empty if there's no matches.

**Comment by shawnc722** [comment]
Do you think it's worth adding a boolean to control whether or not an exact match is needed? I'm imagining something like:
```
def get_devices_by_name(self, name : str, exact : bool=True):
     if exact: return [device for device in self.devices if device.name == name]
     return [device for device in self.devices if name in device.name]
```
This would give it a bit more functionality but the only use cases for it I can see right now are already covered by get_devices_by_type.

**Comment by jath03** [comment]
I like the idea, even if its only benefit is keeping me and other users from having to type in devices' full names.  Also, maybe have the inexact version be case insensitive.

**Comment by shawnc722** [comment]
Alright finally had time to get to this, everything looking good to you?

File: openrgb/orgb.py [hunk 1]
```diff
@@ -307,6 +307,16 @@ def get_devices_by_type(self, type: utils.DeviceType) -> list[Device]:
...
         return [device for device in self.devices if device.type == type]
 
+    def get_devices_by_name(self, name: str, exact: bool = True) -> list[Device]:
+        '''
+        Gets a list of any devices matching the requested name
+        
+        :param name: the name of the device(s) you want to get
+        :param exact: whether to check for only a precise match or accpet a device that contains name
+        '''
+        if exact: return [device for device in self.devices if device.name == name]
```
Comment (jath03) [inline openrgb/orgb.py:317]:
```suggestion
        if exact: 
            return [device for device in self.devices if device.name == name]
```

**Pull request merged**